plugins/sonar-xoo-plugin/build.gradle

```diff
@@ -0,0 +1,39 @@
+apply plugin: 'com.github.johnrengelman.shadow'
+apply plugin: 'maven'
+
+dependencies {
+  compile 'com.google.guava:guava'
+  compile 'commons-io:commons-io'
+  compile 'commons-lang:commons-lang'
+  provided 'com.google.code.findbugs:jsr305'
+  provided project(':sonar-plugin-api')
+
+  testCompile 'junit:junit'
+  testCompile 'org.assertj:assertj-core'
+  testCompile 'org.mockito:mockito-core'
+}
+
+jar {
+  classifier = 'original'
+}
+shadowJar {
+  configurations = [project.configurations.default]
+  // location and name is important for integration tests
+  destinationDir = file('target')
+  classifier = null
+  manifest {
```

Comment:
There is a dedicated plugin to replace sonar-maven-packaging-plugin. Not sure about its quality: https://github.com/iwarapter/gradle-sonar-packaging-plugin